### PR TITLE
OBSOLETE - REFACTORING

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR FEEDBACK

# Summary
- Create Secondary Nav & Variants
- Consolidate Main Nav with Secondary Nav for the drop down
- Allow for multiple levels in the Navigation
- Minor style and behavior changes

# Needed By (Date)
- ?

# Urgency
- ?

# Steps to Test

- [x] Backwards compatibility for TWIG
- [x] Backwards compatibility for SCSS
- [x] Backwards compatibility for JS
- [x] Accessibility on Desktop and mobile for both menu components
- [x] Define deprecated code and plan for removal
- [x] Compatibility with Masthead Component
- [x] Browser compatibility check

# TODO

- [x] find a better place for the main nav expand active path code
- [x] Fix the search variant in main nav as it is missing the search bar.
- [x] Think about the behaviour of the default expanded state when a user resizes their browser for main nav. 
- [x] CSS color changes
- [x] Code climate duplication fixes

# Affected Projects or Products
- All things that used main nav
- Masthead component

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by sherakama** [comment]
@yvonnetangsu To give you time to review and to give me time to react, please have a review of the working code as it is. Please leave all comments you see fit.

**Comment by yvonnetangsu** [comment]
@sherakama , I've pulled down the latest commits from your branch to check for backward compatibility. I noticed a few things (which you might be working on already) for now:

1. Main Nav desktop - If your focus is on a link within a dropdown menu (level 2 link on desktop), if you use the back/forward arrow, it takes you up and down the dropdown menu instead of moving left and right on the first level items (and closing the drop down). See the live Decanter site for expected behavior.
2. Main Nav desktop - The `<li>` that have dropdowns have attribute aria-expanded="false" on them. They shouldn't be there since they should only be applied to `<a>` elements within in this case.
3. Main Nav - "Current" items ("Useful Websites" in our example) aren't styled - missing the border. also, they animate differently when hovered over.

I'll look more closely soon - will add to this page in a different comment. Thank you for working on this!

**Comment by yvonnetangsu** [comment]
Btw, I'd be happy to deal with #3 in the above comment since it's a .scss thing, but wanted to list all the anomalies so we're on the same page 😄

**Comment by sherakama** [comment]
@yvonnetangsu 

Thanks for the comments. Yes, lots of stuff still in flight on the interactivity. But the good new is that you got to that point! I didn't break anything so badly it just failed. Have at the sass fixes and feel free to completely re-jigger the placeholders. I am working through the JS refactor still and will let you know when there is more to see.

**Comment by yvonnetangsu** [comment]
> @yvonnetangsu
> 
> Thanks for the comments. Yes, lots of stuff still in flight on the interactivity. But the good new is that you got to that point! I didn't break anything so badly it just failed. Have at the sass fixes and feel free to completely re-jigger the placeholders. I am working through the JS refactor still and will let you know when there is more to see.

Cool. Yup, compiled no problem. Thanks for letting me know. I'll get to the sass as soon as I can - this week is a little crazy 😂

**Comment by yvonnetangsu** [comment]
@sherakama , I'm starting to work on the style now but I have a couple questions for you - 

1. It looks like main-nav and secondary-nav current shares the same .json data, is this intentional or just a WIP for testing thing? Could I change it to pass in different .json data for main-nav for easier testing?

2. Looks like some modifier classes for nav items are hard coded in the .json files, e.g., `su-secondary-nav__item--expanded`, while in the original main nav, these were automatically added via JS. Is this just for testing?

3. Is the secondary nav going to have the +/- signs or v/^ for the expandable items? I personally think either would work but like to confirm.

Thanks!!

**Comment by yvonnetangsu** [comment]
I'll keep adding discrepancies I observe on this page - 

- In general, spacebar, arrow keys etc are not working properly.

Also - 
Expected behavior when clicking on a parent item (mouse user):
- Focus should stay on the parent item

Expected behavior when hitting enter/spacebar/down arrow on a parent item (keyboard user):
- Focus should move to the first child item (for accessibility), but currently focus stays on the parent item.

**Comment by sherakama** [comment]
> 1. It looks like main-nav and secondary-nav currently share the same .json data, is this intentional or just a WIP for testing thing? Could I change it to pass in different .json data for main-nav for easier testing?

The structure was just wip to see how the secondary nav behavior was working in the main nav. The original main menu schema can be reverted.

> 2. Looks like some modifier classes for nav items are hard coded in the .json files, e.g., `su-secondary-nav__item--expanded`, while in the original main nav, these were automatically added via JS. Is this just for testing?

These are for testing and preview. I will be sure to remove them when the behaviors are worked out.

> 3. Is the secondary nav going to have the +/- signs or v/^ for the expandable items? I personally think either would work but like to confirm.

I think the (+) signs are the way to go for this version. I'd like to continue to add on to this PR to accommodate the accordion button behavior along with the accordion link behavior which I think could then use the carots as their icon.

**Comment by yvonnetangsu** [comment]
@sherakama , thanks for confirming. I've reverted the main nav's json to used the old one in my branch. Feel free to pull it down and take a look.

**Comment by sherakama** [comment]
> * In general, spacebar, arrow keys etc are not working properly.
 
Aye, as part of the refactor I stripped out the controls and are adding them back in attached to the new class objects. As the sidebar, main nav, and button accordion options have slightly different behaviors we will need to be able to set the correct one in the correct place. For this I want to be able to inject stand-alone behavior events on to the correct navitem instance and am working through this. 

> Also -
> Expected behavior when clicking on a parent item (mouse user):
> * Focus should stay on the parent item

Thanks for this note. I am using the current version as what I want to reproduce for the main nav. 

> Expected behavior when hitting enter/spacebar/down arrow (or any key that expand a parent) on a parent item (keyboard user):
> * Focus should move to the first child item (for accessibility) - currently this is not working

I think the behavior for the main nav should be the same but I'd like to find out what the expectation is for the sidebar as it is all vertical and doesn't have the horizontal top level. Ideally it will be the same as the main nav so I'll probably just do it that way and not ask questions :)

**Comment by yvonnetangsu** [comment]
> > * In general, spacebar, arrow keys etc are not working properly.
> 
> Aye, as part of the refactor I stripped out the controls and are adding them back in attached to the new class objects. As the sidebar, main nav, and button accordion options have slightly different behaviors we will need to be able to set the correct one in the correct place. For this I want to be able to inject stand-alone behavior events on to the correct navitem instance and am working through this.
> 
> > Also -
> > Expected behavior when clicking on a parent item (mouse user):
> > 
> > * Focus should stay on the parent item
> 
> Thanks for this note. I am using the current version as what I want to reproduce for the main nav.
> 
> > Expected behavior when hitting enter/spacebar/down arrow (or any key that expand a parent) on a parent item (keyboard user):
> > 
> > * Focus should move to the first child item (for accessibility) - currently this is not working
> 
> I think the behavior for the main nav should be the same but I'd like to find out what the expectation is for the sidebar as it is all vertical and doesn't have the horizontal top level. Ideally it will be the same as the main nav so I'll probably just do it that way and not ask questions :)

Thanks for taking on this one, @sherakama . Really appreciate it 😄 ! Lots of things to think about now that we have so many types of navigations 🤔

**Comment by sherakama** [comment]
@JBCSU & @yvonnetangsu,

I still have a few items to tackle in the checklist but this is in a good place for review feedback. At your leisure.

**Comment by sherakama** [comment]
Since this is going in to 6, I am going to remove the deprecated functions and start a new PR for v5 to mark the existing items as deprecated.

**Comment by yvonnetangsu** [comment]
Love the animation on the +/- toggles 😄

**Comment by yvonnetangsu** [comment]
I see a lot of the hover/focus color is dark-red - I believe we've gotten rid of any use of that color completely in navs, and changed any reference to dark-red to bright-red for high contrast between normal and hover states (for accessibility). See #422

File: .sass-lint.yml [hunk 1]
```diff
@@ -30,7 +32,7 @@ rules:
   leading-zero:
     - 1
     - include: true
-  max-line-length: 120
+  max-line-length: 0
```
Comment (sherakama) [inline .sass-lint.yml]: This is a boolean option from what I gather. Turning off the line length check is better then it being on and set to 80.

File: Gruntfile.js [hunk 2]
```diff
@@ -19,7 +19,6 @@ module.exports = function(grunt) {
           ]
         },
         options: {
-          watchTask: true,
```
Comment (sherakama) [inline Gruntfile.js]: This was conflicting with webpack.

File: kss/builder/decanter/kss-assets/js/main-nav.js [hunk 3]
```diff
@@ -3,26 +3,6 @@
  * @type {[type]}
  */
 (function($) {
-
```
Comment (sherakama) [inline kss/builder/decanter/kss-assets/js/main-nav.js]: All of this functionality has been moved into the nav class now.

File: package.json [hunk 4]
```diff
@@ -33,7 +33,8 @@
     "url": "https://github.com/SU-SWS/decanter/issues"
   },
   "dependencies": {
-    "bourbon": "^5.1.0"
+    "bourbon": "^5.1.0",
+    "custom-event-polyfill": "^1.0.7"
```
Comment (sherakama) [inline package.json]: Used to add data to the custom events.
https://www.npmjs.com/package/custom-event-polyfill

File: .codeclimate.yml [hunk 5]
```diff
@@ -1,5 +1,12 @@
 version: "2"
-engines:
+checks:
```
Comment (sherakama) [inline .codeclimate.yml]: These thresholds come from our developer discussions and are set in our default boilerplate module: https://github.com/SU-SWS/stanford_module_example/blob/8.x-2.x/.codeclimate.yml#L2

**Comment by yvonnetangsu** [comment]
Looking better and better! 😄 
This is a design question so likely would involve @kerri-augenstein also - what do you think about making the circle toggles a bit smaller (40px-ish)? I'm not too crazy about the jagged 1px border on the light version - could we either do 2px thick border or use a background color instead?

**Comment by yvonnetangsu** [comment]
You probably know these things already but just to make sure we're on the same page - 

- I notice on the horizontal desktop menu, as you hover on a link, the transformation on the bar is vertical but it should be horizontal (the bar would expand to left and right from the center). 
- When one opens up a parent menu using the mouse, focus doesn't move automatically to first children link. Focus only move to first link for keyboard users when they use keyboard to expand a menu.

**Comment by kerri-augenstein** [comment]
@yvonnetangsu 

1. I like the idea of making the circles smaller as well. If it's an easy tweak, would love to get that in. 40px sounds great. And then it might need re-centering the button vertically. 

2. For the style, what about a background-color: #f4f4f4, and drop the outline? I'm wondering if that will be more of a visual indication to the user to "tap". 

3. Also, I'm noticing that when I hover on the expand button (circle version), there is a flicker of a hoverstate appearing on the word (underline appears) and the border-left flickers in as well.

**Comment by yvonnetangsu** [comment]
> @yvonnetangsu
> 
> 1. I like the idea of making the circles smaller as well. If it's an easy tweak, would love to get that in. 40px sounds great. And then it might need re-centering the button vertically.

Cool. Yeah, recentering and some margin on the right would be nice.

> 2. For the style, what about a background-color: #f4f4f4, and drop the outline? I'm wondering if that will be more of a visual indication to the user to "tap".

I'm only worried about the color contrast of #f4f4f4 with white background. What if we use a slightly darker grey? Is that too distracting when place on a sidebar? I agree that having a background color is more of a visual indication to tap.

> 3. Also, I'm noticing that when I hover on the expand button (circle version), there is a flicker of a hoverstate appearing on the word (underline appears) and the border-left flickers in as well.

I don't notice the flicker - but I know what you mean, I think it's due to the cursor moving from outside the circle (but inside the text link area) into the circle (since the circle doesn't extend all the way to the edge).

**Comment by sherakama** [comment]
That flicker is caused by the fact that the button is over the link element and you have to hover over the link before you get to the button. I have added some margin to the right side so this effect should be clearer now but perhaps the link will have to be shorter. Not sure how that is going to play out yet.

**Comment by sherakama** [comment]
# TODO

[ ] Flicker on hover (link vs toggle)
[ ] Hover transition on main menu horizontal (expand horizontally)
[ ] Update v6 and master branches prior to merge
[ ] Lower the z-index
[ ] When one opens up a parent menu using the mouse, focus doesn't move automatically to first children link. Focus only move to first link for keyboard users when they use keyboard to expand a menu.
[ ] Color contrast check

**Comment by sherakama** [comment]
I ran the color contrast checker in chrome and it looks like of the text passes but the active (bright red) highlight on the dark background does not. We may want to choose another color?

contrast-wcag-aa-small (1)

**Comment by yvonnetangsu** [comment]
> I ran the color contrast checker in chrome and it looks like all of the text passes but the active (bright red) highlight on the dark background does not. We may want to choose another color?
> 

@sherakama , I remember John Holleman checked the contrast and it was ok. Probably because he ran the test for bold/large text since the border is thick. The contrast requirement for bold/large text is less strict.

**Comment by sherakama** [comment]
Thanks Yvonne,

I re-ran the check with the larger text requirement and it passed. Going forward with your blessing.

Also, I think I have checked all the to-dos off now. Please have a review.

**Comment by yvonnetangsu** [comment]
> Thanks Yvonne,
> 
> I re-ran the check with the larger text requirement and it passed. Going forward with your blessing.
> 
> Also, I think I have checked all the to-dos off now. Please have a review.

Wow you're so fast. You even did the toggle active area thing! I'll review as soon as I get a chance.

**Comment by yvonnetangsu** [comment]
Getting really really close! Found one thing:

- When you're on desktop main nav, if your focus in in the dropdown (2nd level links), the right arrow works perfectly, but the left arrow doesn't move your to the previous parent (you have to hit twice).

**Comment by sherakama** [comment]
@yvonnetangsu Ok, I believe I have the left arrow behavior correct now. Please have another pass.

**Comment by yvonnetangsu** [comment]
> @yvonnetangsu Ok, I believe I have the left arrow behavior correct now. Please have another pass.

Hi @sherakama , I had another look. Left button works perfectly now. Thanks!

I made a small change to remove the `aria-expanded` property on `<nav>` - please see this ticket from the Level Access audit. It doesn't seem to impact other things but feel free to tweak it if you see anything weird. https://github.com/SU-SWS/decanter/issues/396

I noticed a couple more minor issues. Will add after this comment.

**Comment by yvonnetangsu** [comment]
- On the mobile menu, I can't seem to tab to the site search within the menu. Currently on the live site, it's the last item that someone can tab to on a mobile menu.

- When focus is on a link inside a submenu, hitting the ESC key should close that level of submenu first, then return focus to the parent link. For example, if I"m on a 4th level menu link, hitting ESC once will close the 4th level menu, hitting it again will close the 3rd.....etc. until all child menus are close, then the last ESC click will close the whole menu (on main nav will return focus to the hamburger).

**Comment by sherakama** [comment]
I'll look into the search box tab handling. 

For the esc behavior, where does this recommendation come from? I've not seen that pattern before.

**Comment by yvonnetangsu** [comment]
> I'll look into the search box tab handling.
> 
> For the esc behavior, where does this recommendation come from? I've not seen that pattern before.

Thanks, @sherakama !

The ESC comes from here - 
https://www.w3.org/TR/wai-aria-practices/#menu

`Escape: Close the menu that contains focus and return focus to the element or context, e.g., menu button or parent menuitem, from which the menu was opened.`

I know Kevin did that for the Identity site so I adopted that behavior and it seems like that's what  the w3 link recommended also.

**Comment by sherakama** [comment]
Awesome, thanks for the link. I'll socialize this with others.

**Comment by sherakama** [comment]
Ok, revisions are in. I am not proud of the solution for closing the menu when `shift + tab` out of the menu with the mobile search but it works. Please have another review.

**Comment by yvonnetangsu** [comment]
Here's one more:

**This is only about when both parent/children are vertical menu items (exclude Main Nav 1st/2nd level as that works perfectly now):**
Let's say my focus is on the item with left white bar in the below screenshot
Screen Shot 2019-08-28 at 2 40 19 PM

- If I hit the left arrow, it should close that submenu, and return focus to "Current - first level item" (the parent).
- If I hit the right arrow, it should do nothing.
- Note: this is different from on Main Nav desktop version, where the parent level is a horizontal menu.

**Comment by yvonnetangsu** [comment]
Btw, @sherakama , if you feel like it would be a lot easier to have some special sauce code for the Main Nav or Secondary Nav, please feel free to do so. It's nice to have reusable code but only if it doesn't take too much extra time.

**Comment by sherakama** [comment]
S'all good. This refactor is really widdling it down to just the differences that are needed. Depending on how many there are I may refactor the MainEvents.js file into multiple classes specifically for the main nav component as the idea of that file was to inject minor behavior differences.

**Comment by yvonnetangsu** [comment]
More arrow behavior for documentation - 

Again, this is only about **when both parent/children are vertical menu items** (exclude Main Nav 1st/2nd level as that works perfectly now):
Let's say my focus is on a parent item (with left white bar in the below screenshot) - 

Screen Shot 2019-08-28 at 5 03 35 PM

- If I hit the left arrow, it should do nothing.
- Right arrow behavior is perfect now (hitting right arrow opens up the menu and place focus on first child item)

**Comment by sherakama** [comment]
@yvonnetangsu Ok, So I made a mess of this last night. 

The event injection was swapped out for an inheritance model to support the ongoing complexities of the main nav vs the secondary nav, and I was successful in getting the desired behavior. Unfortunately, with the way the inheritance is working, I ended up with code duplication.  

Code climate is complaining about the duplication and it will need some love but that will just have to wait until I get back unless somebody else wants to pick it up and run with it. 

After handling the refactor and looking back, I suspect the inheritance model isn't the best fist and some format of composition is needed again.

**Comment by yvonnetangsu** [comment]
@sherakama , thank you for your work on this! Enjoy your vacation, and I'll look at it while you're away 😸

**Comment by yvonnetangsu** [comment]
@sherakama , noticed a couple things (more in-depth review coming) for now:

- Main Nav, desktop version: Let's say a subnav is already expanded. If I click on a different parent item, it should first close all other open subnav(s), before it expands the current subnav (of the parent that I'm clicking).
- Main Nav, desktop version: since the auto expansion to expose current active item doesn't include expanding the subnav (2nd level), we need the active class added to the top level parent li of the active item.

**Comment by sherakama** [comment]
Should the main nav be expanded by default or just marked with the active class?

**Comment by yvonnetangsu** [comment]
> Should the main nav be expanded by default or just marked with the active class?

Perhaps the dropdown menus are collapsed by default, but once you click on a 1st level parent to  expand a particular dropdown, if the active page is buried in say, the 3rd level, then the dropdown should be expanded to expose the 3rd level active? I could be convinced otherwise. On the UComm site, we most likely would avoid having more than 2 levels of menu links so this probably affects SWS more.

**Comment by sherakama** [comment]
Yeah, let me sleep on this and circle up with Kerri tomorrow.

**Comment by sherakama** [comment]
@yvonnetangsu,
@kerri-augenstein, 

I've added a new main-nav toggle option as it looks like we might need something like it. It needs work on the keyboard controls and I will review with Kerri shortly. Any comments or help you can provide is welcome as usual.

**Comment by sherakama** [comment]
@yvonnetangsu & @kerri-augenstein 

As this has been one of our longest open tickets I'd like to set the acceptance criteria with you for getting it merged into V6 as "good enough" knowing that we can continue to work on outstanding minor issues. 

So I propose the following criteria to be met before merging:

[ ] - Main nav accordion style keyboard controls pass @yvonnetangsu's critique
[ ] - Main nav button style keyboard controls closely match accordion style
[ ] - Refactor placeholders to mixins or duplication
[ ] - Visual styles are "good enough" by @kerri-augenstein's critique
[ ] - Active class expand to open approved by @yvonnetangsu 

Anything else you would like to add?

**Comment by yvonnetangsu** [comment]
@sherakama , that sounds like a good plan. I'll find sometime to review this as soon as I can. "Main nav button style keyboard controls closely match accordion style" - does that mean the main nav desktop styles (the horizontal menu)?

**Comment by sherakama** [comment]
> @sherakama , that sounds like a good plan. I'll find sometime to review this as soon as I can. "Main nav button style keyboard controls closely match accordion style" - does that mean the main nav desktop styles (the horizontal menu)?

I would say it should be on both mobile and desktop

**Comment by yvonnetangsu** [comment]
> @yvonnetangsu,
> @kerri-augenstein,
> 
> I've added a new main-nav toggle option as it looks like we might need something like it. It needs work on the keyboard controls and I will review with Kerri shortly. Any comments or help you can provide is welcome as usual.

I think I missed this comment. Will look at it this week.

**Comment by sherakama** [comment]
Well here is some more fun. I've got preview environments linked up. Check out `.su-main-nav--buttons`

https://pr443-hqb3gxjhkjjrphixmzvl7wnkq2rgeatm.tugboat.qa/section-components.html#kssref-components-main-navigation

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-hqb3gxjhkjjrphixmzvl7wnkq2rgeatm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5da8a8ad44c5fa006cea9107

**Comment by yvonnetangsu** [comment]
@sherakama , is having the main nav (accordion/original) styles match what we have in master in scope for this PR? 

For the main nav button styles, could you add focus state for the caret buttons (make it same as hover state - ideally adding a color change on top of the size change would be nice also, for both hover and focus)? That would make it easier to test the keyboard controls. Thanks!

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-hqb3gxjhkjjrphixmzvl7wnkq2rgeatm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5da8a8ad44c5fa006cea9107

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-hqb3gxjhkjjrphixmzvl7wnkq2rgeatm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5da8a8ad44c5fa006cea9107

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-hqb3gxjhkjjrphixmzvl7wnkq2rgeatm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5da8a8ad44c5fa006cea9107

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-hqb3gxjhkjjrphixmzvl7wnkq2rgeatm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5da8a8ad44c5fa006cea9107

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-fczdwxwdmkgii6jp0ngwcuxcljyb0k6d.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5daf976944c5fa0b1b088723

Visual Diffs:
- https://dashboard.tugboat.qa/5daf976944c5fa0b1b088723/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-fczdwxwdmkgii6jp0ngwcuxcljyb0k6d.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5daf976944c5fa0b1b088723

Visual Diffs:
- https://dashboard.tugboat.qa/5daf976944c5fa0b1b088723/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-fczdwxwdmkgii6jp0ngwcuxcljyb0k6d.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5daf976944c5fa0b1b088723

Visual Diffs:
- https://dashboard.tugboat.qa/5daf976944c5fa0b1b088723/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-fczdwxwdmkgii6jp0ngwcuxcljyb0k6d.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5daf976944c5fa0b1b088723

Visual Diffs:
- https://dashboard.tugboat.qa/5daf976944c5fa0b1b088723/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-fczdwxwdmkgii6jp0ngwcuxcljyb0k6d.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5daf976944c5fa0b1b088723

Visual Diffs:
- https://dashboard.tugboat.qa/5daf976944c5fa0b1b088723/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-nq0b5qhheg58go9hwdxhd2v8yyb4dyoo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5db0823a44c5fa66670d9f88

Visual Diffs:
- https://dashboard.tugboat.qa/5db0823a44c5fa66670d9f88/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-jaezsl8ffx5ycvhvjvtzblzct0dvak28.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5db087ac44c5fa575d0de0f9

Visual Diffs:
- https://dashboard.tugboat.qa/5db087ac44c5fa575d0de0f9/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-jaezsl8ffx5ycvhvjvtzblzct0dvak28.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5db087ac44c5fa575d0de0f9

Visual Diffs:
- https://dashboard.tugboat.qa/5db087ac44c5fa575d0de0f9/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-jaezsl8ffx5ycvhvjvtzblzct0dvak28.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5db087ac44c5fa575d0de0f9

Visual Diffs:
- https://dashboard.tugboat.qa/5db087ac44c5fa575d0de0f9/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-jaezsl8ffx5ycvhvjvtzblzct0dvak28.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5db087ac44c5fa575d0de0f9

Visual Diffs:
- https://dashboard.tugboat.qa/5db087ac44c5fa575d0de0f9/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-jaezsl8ffx5ycvhvjvtzblzct0dvak28.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5db087ac44c5fa575d0de0f9

Visual Diffs:
- https://dashboard.tugboat.qa/5db087ac44c5fa575d0de0f9/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-jaezsl8ffx5ycvhvjvtzblzct0dvak28.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5db087ac44c5fa575d0de0f9

Visual Diffs:
- https://dashboard.tugboat.qa/5db087ac44c5fa575d0de0f9/visualdiffs

**Comment by yvonnetangsu** [comment]
@sherakama , I had a chance to talk to @kerri-augenstein last week. I have a question about the `.su-main-nav--buttons` variant. Kerri said she's not aware of the desktop behavior of this variant (where the caret button is larger on hover/focus). She said the desktop variant basically function like the no-js version of the default desktop Main Nav. Basically if you click on the parent item, it takes you to the site it links to, but the drop down expands once you hover over the parent item as well. She said the split link/button format is only for the mobile/vertical menu version. Just bringing this up to make sure we're all on the same page.

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-jaezsl8ffx5ycvhvjvtzblzct0dvak28.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5db087ac44c5fa575d0de0f9

Visual Diffs:
- https://dashboard.tugboat.qa/5db087ac44c5fa575d0de0f9/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr443-jaezsl8ffx5ycvhvjvtzblzct0dvak28.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5db087ac44c5fa575d0de0f9

Visual Diffs:
- https://dashboard.tugboat.qa/5db087ac44c5fa575d0de0f9/visualdiffs

**Comment by sherakama** [comment]
I'm going to leave the branch just a little while longer.

**Pull request closed without merging**